clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/subscriber/PubSubSubscriberClientImpl.java

```diff
@@ -75,27 +75,30 @@ public PubSubSubscriber(
     /**
      * Subscribes messages from Pub/Sub.
      *
+     * @param keepTrying Condition that controls whether another attempt to subscribe to a subscription should be
+     *            started. Client code can set this to {@code false} in order to prevent further attempts.
      * @return A future indicating the outcome of the operation.
-     * @throws ServerErrorException If subscribing was not successful.
      */
-    public Future<Void> subscribe() {
-        try {
-            subscriber.addListener(
-                    new ApiService.Listener() {
+    public Future<Void> subscribe(final boolean keepTrying) {
+        final Promise<Void> resultPromise = Promise.promise();
+        subscribeWithRetries(resultPromise, keepTrying);
+        return resultPromise.future();
+    }
 
-                        @Override
-                        public void failed(final ApiService.State from, final Throwable failure) {
-                            log.error("Error subscribing message from Pub/Sub", failure);
-                            throw new ServerErrorException(HttpURLConnection.HTTP_UNAVAILABLE,
-                                    "Error subscribing message from Pub/Sub", failure);
-                        }
-                    },
-                    MoreExecutors.directExecutor());
+    private void subscribeWithRetries(final Promise<Void> resultPromise, final boolean keepTrying) {
+        try {
             subscriber.startAsync().awaitRunning();
-            return Future.succeededFuture();
-        } catch (IllegalStateException e) {
-            log.error("Service reached illegal state", e);
-            return Future.failedFuture(e);
+            log.info("Successfully subscribing on: {}", subscriber.getSubscriptionNameString());
+            resultPromise.complete();
+        } catch (Exception e) {
+            if (keepTrying) {
+                log.error("Error subscribing message from Pub/Sub, will retry in {}ms: ", SUBSCRIBE_RETRY_DELAY_MILLIS,
```

Comment:
I do not think that this should be logged as an error. In the end, we are actually expecting this to happen and are dealing with it accordingly, are we not? So, logging at info level should be more than enough FMPOV ...

Reply:
Yes, you're right, updated to info level